scripts/make_statistics.py

```diff
@@ -0,0 +1,539 @@
+            [n_elem], columns=_multi_index_scen(rulename, [header + "-size"])
+        )
+
+    else:
+        return pd.DataFrame()
+
+
+def collect_network_osm_stats(path, rulename, header, metric_crs="EPSG:3857"):
+    """
+    Collect statistics on OSM network data:
+    - number of items
+    - length of the stored shapes
+    - length of objects with tag_frequency == 0 (DC elements)
+    """
+
+    if os.path.exists(path) and (os.stat(path).st_size > 0):
+        df = gpd.read_file(path)
+        n_elem = len(df)
+        obj_length = (
+            df["geometry"].apply(make_valid).to_crs(crs=metric_crs).geometry.length
+        )
+        len_obj = np.nansum(obj_length * df.circuits)
+
+        len_dc_obj = 0.0
+        n_elem_dc = 0
+        if "tag_frequency" in df.columns:
+            coerced_vals = pd.to_numeric(df.tag_frequency, errors="coerce")
+            coerced_vals.fillna(50, inplace=True)
+            idx_dc = coerced_vals[coerced_vals.astype(int) == 0].index
+            len_dc_obj = np.nansum(obj_length.loc[idx_dc] * df.circuits.loc[idx_dc])
+            n_elem_dc = int(len(idx_dc))
+
+        return pd.DataFrame(
+            [[n_elem, n_elem_dc, len_obj, len_dc_obj]],
+            columns=_multi_index_scen(
+                rulename,
+                [header + "-" + k for k in ["size", "size_dc", "length", "length_dc"]],
+            ),
+        )
+    else:
+        return pd.DataFrame()
+
+
+def collect_osm_stats(rulename, **kwargs):
+    """
+    Collect statistics on OSM data.
+    When lines and cables are considered, then network-related statistics are collected
+    (collect_network_osm_stats), otherwise basic statistics are (collect_basic_osm_stats)
+    """
+
+    metric_crs = kwargs.pop("metric_crs", "EPSG:3857")
+    only_basic = kwargs.pop("only_basic", False)
+
+    df_list = []
+
+    for k, v in kwargs.items():
+        if not only_basic and (k in ["lines", "cables"]):
+            df_list.append(
+                collect_network_osm_stats(v, rulename, k, metric_crs=metric_crs)
+            )
+        else:
+            df_list.append(collect_basic_osm_stats(v, rulename, k))
+
+    return pd.concat(df_list, axis=1)
+
+
+def collect_raw_osm_stats(rulename="download_osm_data", metric_crs="EPSG:3857"):
+    """
+    Collect basic statistics on OSM data; used for raw OSM data.
+    """
+
+    snakemake = _mock_snakemake("download_osm_data")
+
+    options_raw = dict(snakemake.output)
+    options_raw.pop("generators_csv")
+
+    df_raw_osm_stats = collect_osm_stats(
+        rulename, only_basic=True, metric_crs=metric_crs, **options_raw
+    )
+
+    add_computational_stats(df_raw_osm_stats, snakemake)
+
+    return df_raw_osm_stats
+
+
+def collect_clean_osm_stats(rulename="clean_osm_data", metric_crs="EPSG:3857"):
+    """
+    Collect statistics on OSM data; used for clean OSM data.
+    """
+
+    snakemake = _mock_snakemake("clean_osm_data")
+
+    options_clean = dict(snakemake.output)
```

Comment:
Same snakemake comment as above.. Please review the function

Reply:
Same as above